include/aspect/particle/generator/interface.h

```diff
@@ -0,0 +1,161 @@
+/*
+ Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_generator_interface_h
+#define __aspect__particle_generator_interface_h
+
+#include <aspect/particle/world.h>
+#include <aspect/particle/particle.h>
+#include <aspect/plugins.h>
+
+#include <deal.II/base/parameter_handler.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      /**
+       * Abstract base class used for classes that generate particles
+       */
+      template <int dim>
+      class Interface
+      {
+        public:
+          /**
+           * Constructor.
+           */
+          Interface() {}
+
+          /**
+           * Destructor. Made virtual so that derived classes can be created
+           * and destroyed through pointers to the base class.
+           */
+          virtual ~Interface () {}
+
+          /**
+           * Generate a specified number of particles in the specified world
+           * using the type of generation function implemented by this
+           * Generator.
+           *
+           * @param [in] total_num_particles Total number of particles to
+           * generate. The actual number of generated particles may differ,
+           * for example if the generator reads particles from a file this
+           * parameter may be ignored.
```

Comment:
This parameter has disappeared from the signature of the function. Please update the documentation (including the first paragraph). Incidentally, how do generators know how many particles to declare? Through their run-time parameters? I think this would be worthwhile documenting with this function.